File: DocumentAnalysisSystem/Readers/PdfDocumentReader.cs
using System.IO;
using System.Linq;
using DocumentAnalysisSystem.Models;
using UglyToad.PdfPig;

namespace DocumentAnalysisSystem.Readers
{
    public class PdfDocumentReader : IDocumentReader
    {
        public Document ReadDocument(string filePath)
        {
            var document = new Document(Path.GetFileName(filePath));
            using var pdf = PdfDocument.Open(filePath);

            var textBuilder = new System.Text.StringBuilder();
            foreach (var page in pdf.GetPages())
            {
                textBuilder.AppendLine(page.Text);
                textBuilder.AppendLine();
            }

            var paragraphs = textBuilder.ToString().Split(new[] { "\r\n\r\n", "\n\n" }, System.StringSplitOptions.RemoveEmptyEntries);
            foreach (var paragraphText in paragraphs)
            {
                var text = paragraphText.Trim();
                if (!string.IsNullOrEmpty(text))
                {
                    document.Add(new Paragraph(text));
                }
            }

            return document;
        }
    }
}

File: DocumentAnalysisSystemJava/src/main/java/com/documentanalysis/readers/PdfDocumentReader.java
package com.documentanalysis.readers;

import com.documentanalysis.models.Document;
import com.documentanalysis.models.Paragraph;
import org.apache.pdfbox.pdmodel.PDDocument;
import org.apache.pdfbox.text.PDFTextStripper;

import java.io.File;
import java.nio.file.Paths;

public class PdfDocumentReader implements IDocumentReader {
    @Override
    public Document readDocument(String filePath) throws Exception {
        Document document = new Document(Paths.get(filePath).getFileName().toString());

        try (PDDocument pdf = PDDocument.load(new File(filePath))) {
            PDFTextStripper stripper = new PDFTextStripper();
            String text = stripper.getText(pdf);
            String[] paragraphs = text.split("(?:\r\n){2,}|(?:\n){2,}");

            for (String paragraphText : paragraphs) {
                String trimmed = paragraphText.trim();
                if (!trimmed.isEmpty()) {
                    document.add(new Paragraph(trimmed));
                }
            }
        }

        return document;
    }
}
